File: hrpay/api/department.py
import frappe
import json

@frappe.whitelist()
def get_departments():
    try:
        departments = frappe.get_all("Department", fields=["name", "department_name"])
        return json.dumps(departments)
    except Exception as e:
        frappe.log_error(f"Error fetching departments: {str(e)}")
        return json.dumps({"error": str(e)})

File: hrpay/api/department.js
export async function fetchDepartments() {
  try {
    const response = await fetch("/api/method/hrpay.api.get_departments");
    const data = await response.json();
    return data.map(dept => ({ id: dept.name, name: dept.department_name }));
  } catch (error) {
    console.error("Error fetching departments:", error);
    return [];
  }
}